src/NUnitEngine/nunit.engine.tests/Internal/DirectoryFinderTests.cs

```diff
@@ -19,66 +19,658 @@
 // LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 // OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-// ***********************************************************************using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Text;
-using NUnit.Framework;
+// ***********************************************************************
 
 namespace NUnit.Engine.Internal.Tests
 {
+    using NSubstitute;
+    using NUnit.Engine.Internal.FileSystemAccess;
+    using NUnit.Framework;
+    using System.Collections.Generic;
+    using System.Linq;
+    using SIO = System.IO;
+
```

Comment:
Last nit-pick :). These should be moved outside. Besides this the rest looks great. Again thanks for the contribution. We need a second review before merging this.